app/Http/Controllers/User/UsersController.php

```diff
@@ -284,14 +285,22 @@ private function loginWithSocialProvider($provider_id, $provider_user_id, $reque
             ]);
         }
 
-        // Link the social account provider
```

Comment:
this logic still is not quite right.  The query is right... query Account with user_id, provider_id and project_id.  But firstOrCreate is not correct in the situation where the provider_user_id in the request does not match what was in the DB. If a record is found from the query, then compare provider_user_id to what was passed in. If they match, then all is good. If they differ, then update Account